bigint_func.circom

```diff
@@ -0,0 +1,442 @@
+pragma circom 2.0.2;
+
+function isNegative(x) {
+    // half babyjubjub field size
+    return x > 10944121435919637611123202872628637544274182200208017171849102093287904247808 ? 1 : 0;
+}
+
+function div_ceil(m, n) {
+    var ret = 0;
+    if (m % n == 0) {
+        ret = m \ n;
+    } else {
+        ret = m \ n + 1;
+    }
+    return ret;
+}
+
+function log_ceil(n) {
+   var n_temp = n;
+   for (var i = 0; i < 254; i++) {
+       if (n_temp == 0) {
+          return i;
+       }
+       n_temp = n_temp \ 2;
+   }
+   return 254;
+}
+
+function SplitFn(in, n, m) {
+    return [in % (1 << n), (in \ (1 << n)) % (1 << m)];
+}
+
+function SplitThreeFn(in, n, m, k) {
+    return [in % (1 << n), (in \ (1 << n)) % (1 << m), (in \ (1 << n + m)) % (1 << k)];
+}
+
+// in is an m bit number
+// split into ceil(m/n) n-bit registers
+function splitOverflowedRegister(m, n, in) {
```

Comment:
Assumptions:
1. in is an m bit number (as already stated in comments)
2. div_ceil(m, n) <= 100 (if m == 252, then n > 2)